tests/test_visibility.py

```diff
@@ -301,22 +299,26 @@ class _TestStore:
 
     """
 
-    def __init__(self):
+    def __init__(self) -> None:
         # data for get_events: a map from event_id to event
-        self.events = {}
+        self.events: Dict[str, EventBase] = {}
 
         # data for get_state_ids_for_events mock: a map from event_id to
-        # a map from (type_state_key) -> event_id for the state at that
+        # a map from (type, state_key) -> event_id for the state at that
         # event
-        self.state_ids_for_events = {}
+        self.state_ids_for_events: Dict[str, Dict[Tuple[str, Optional[str]], str]] = {}
```

Comment:
Is `(Mutable)StateMap[str]` appropriate for the types in this file?

Also does the `Optional[str]` need to be optional?